src/main/java/edu/agh/wfiis/designpatterns/patternsUsageExample/DesignPatternsManager.java

```diff
@@ -1,79 +1,75 @@
+import edu.agh.wfiis.designpatterns.patternsUsageExample.generator.DesignPatternsCourseworkGenerator;
 
 import java.io.File;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
 
 public class DesignPatternsManager {
 
-    public DesignPattern create(PatternName patternName) {
-        switch (patternName) {
-            case BUILDER: {
-                DesignPattern pattern = new DesignPattern("1","builder","builder pattern",null, true, false, false);
-                return pattern;
-            }
-            case DECORATOR: {
-                DesignPattern pattern = new DesignPattern("2","decorator","decorator pattern",null, true, false, false);
-                return pattern;
-            }
-            case STRATEGY: {
-                DesignPattern pattern = new DesignPattern("3","strategy","strategy pattern",null, false, false, true);
-                return pattern;
-            }
-            default: {
-                DesignPattern pattern = new UndefinedPattern();
-                return pattern;
-            }
-        }
+    private final BadCodeExtractor fileExtractor = new FromFileExtractor();
+    private final BadCodeExtractor classLoaderExtractor = new FromClassLoaderExtractor();
+    private final CourseworkGenerator courseworkGenerator = new DesignPatternsCourseworkGenerator();
+
+
+    public DesignPattern create(DesignPatternFactory designPatternFactory) {
+        return designPatternFactory.getDesignPattern();
     }
 
-    public String provideUsageExample(DesignPattern pattern){
-        String example = pattern instanceof UndefinedPattern? null: pattern.getExample();
-        return example;
+    public String provideUsageExample(DesignPattern pattern) {
+        return pattern.getExample();
     }
 
-    public String refactorCode(DesignPattern pattern, String badCode){
-        String goodCode = pattern instanceof UndefinedPattern? null: pattern.refactor(badCode);
-        return goodCode;
+    public String refactorCode(Refactorable pattern, String badCode) {
+        return pattern.refactor(badCode);
     }
 
-    public String listPatterns(){
-        return PatternName.values().toString();
+    public String listPatterns(Set<DesignPatternFactory> designPatternFactorySet) {
+        return String.join(", ", designPatternFactorySet
+                .stream()
+                .map(designPatternFactory -> designPatternFactory.getDesignPattern().getName())
+                .collect(Collectors.toList()));
     }
 
-    public boolean areSameType(DesignPattern pattern1, DesignPattern pattern2){
-        return pattern1.isBehavioral() == pattern2.isBehavioral() && pattern1.isCreational() == pattern2.isCreational() && pattern1.isStructural() == pattern2.isStructural();
+    public boolean areSameType(DesignPattern pattern1, DesignPattern pattern2) {
+        return pattern1.isSameType(pattern2);
```

Comment:
Argummenting for the applying the SOA rules - DesignPattern is a DTO and shouldn't contain any functionality. 

The whole functionality should be contained in this method.